graspologic/plot/plot.py

```diff
@@ -671,6 +673,229 @@ def pairplot(
+        elements.
+    alpha : float, optional, default: 0.7
+        Opacity value of plotter markers between 0 and 1
+    figsize : tuple
+        the size of the 2d subplots configuration
+    kde : boolean
+        allows for access of KDE argument in histplot
+
+    See Also
+    --------
+    graspologic.cluster.AutoGMMCluster
+    sklearn.mixture.GaussianMixture
+    """
+    # Handle X and labels
+    if labels is not None:
+        check_X_y(X, labels)
+    # Handle gmm
+    if gmm is None:
+        msg = "You must input a sklearn.mixture.GaussianMixture"
+        raise NameError(msg)
+    Y_, means, covariances = gmm.predict(X), gmm.means_, gmm.covariances_
+    data = pd.DataFrame(data=X)
+    n_components = gmm.n_components
+    # setting up the data DataFrame
+    if labels is None:
+        lab_names = [i for i in range(n_components)]
+        data["labels"] = np.asarray([lab_names[Y_[i]] for i in range(Y_.shape[0])])
+    else:
+        data["labels"] = labels
+    data["clusters"] = Y_
+    # labels are given we must check whether input is correct
+    if labels is not None:
+        if isinstance(label_palette, str):
+            colors = sns.color_palette(label_palette, n_components)
+            label_palette = dict(zip(np.unique(np.asarray(labels)), colors))
+        elif not isinstance(label_palette, dict):
+            msg = "When giving labels must supply palette in string or dictionary"
+            raise ValueError(msg)
+        if isinstance(cluster_palette, str):
+            colors = sns.color_palette(cluster_palette, n_components)
+            cluster_palette = dict(zip(np.unique(Y_), colors))
+        elif not isinstance(label_palette, dict):
+            msg = "When giving labels must supply palette in string or dictionary"
+            raise ValueError(msg)
+    else:
+        # no labels given we go to default.
+        colors = sns.color_palette(cluster_palette, n_components)
+        labels = np.unique(Y_)
+        cluster_palette = dict(zip(np.unique(labels), colors))
+        label_palette = dict(zip(np.unique(np.asarray(labels)), colors))
+
+    with sns.plotting_context(context=context, font_scale=font_scale):
+        dimensions = X.shape[1]
+        # we only want 1 scatter plot for 2 features
+        if X.shape[1] == 2:
+            dimensions = 1
+        pairplot, axes = plt.subplots(
+            dimensions, dimensions, figsize=figsize, squeeze=False
+        )
+        # this will allow for uniform iteration whether axes was 2d or 1d
+        axes = axes.flatten()
+        for i in range(dimensions):
+            for j in range(dimensions):
+                if i == j and X.shape[1] > 2:
+                    # take care of the distplot on diagonal
+                    for t, lab in zip([i for i in range(X.shape[1])], label_palette):
+                        sns.distplot(
+                            X[Y_ == t, i],
+                            kde=kde,
```

Comment:
what I meant was not just the KDE argument specifically, but rather, letting the user possibly pass a dictionary of kwargs that will get passed to this function

Reply:
does this make sense?

Reply:
@bdpedigo I am a little confused on how that would work

Reply:
https://github.com/microsoft/graspologic/blob/38c70866bdc1b35a3717ceb759f37f1851a67877/graspologic/models/sbm.py#L147 this help?

Reply:
a little? but wouldn't that be cumbersome? because i am unsure what else the use would pass in aside from that? and from my perspective, i would just access these kwargs like kwargs['kde'] anyway. like i would not use kwargs elsewhere. @bdpedigo. Unless i am not understanding something about histplot?

Reply:
https://github.com/microsoft/graspologic/pull/541#discussion_r532812075

I have been talking about distplot/histplot (see my original comment above). not the kde_kws like https://github.com/microsoft/graspologic/pull/541/commits/18f99eb48ffe871cd1022f359d568295e14795b3

Reply:
so i agree that kde_kws is overkill and not what you want, but I was just asking for `histplot_kws` basically.

Reply:
@bdpedigo ok. Let me try that then!

Reply:
@bdpedigo does this work? https://github.com/microsoft/graspologic/pull/541/commits/8c7e71ba2a46fa16fcffeae6dad323bdf7bb8720